[HDBDD] Association create statement not processed correctly

### Details

```
  entity Customers {
      key Id               : String(32);
      Username             : String(100);
      FirstName            : String(500);
      LastName             : String(500);
      Phone                : String(200);
      Email                : String(300);
      ShippingAddress      : association to Products.ShippingAddresses { Id };
   };

   entity ShippingAddresses {
         key Id             : String(32);
         Country            : CountryType;
         City               : String(32);
         PostalCode         : String(32);
   };
```

Results in
`CREATE COLUMN TABLE "XSK_SAMPLES_PRODUCTS"."products.db::Products.Customers" ( "Id" NVARCHAR (32) PRIMARY KEY , "Username" NVARCHAR (100) , "FirstName" NVARCHAR (500) , "LastName" NVARCHAR (500) , "Phone" NVARCHAR (200) , "Email" NVARCHAR (300) , "ShippingAddress.Id" NVARCHAR (32) , CONSTRAINT "products"."db::Products"."Customers"."ShippingAddress" FOREIGN KEY ( "ShippingAddress.Id" ) REFERENCES "XSK_SAMPLES_PRODUCTS"."products.db::Products.ShippingAddresses"( "Id" ))`

Which is not valid and returns an error 
```
2022-02-25 09:01:40.148 [ERROR] [pool-498-thread-1] c.s.x.h.d.f.XSKHDBCoreSynchronizationFacade - SAP DBTech JDBC: [257]: sql syntax error: incorrect syntax near ".": line 1 col 306 (at pos 306)
com.sap.db.jdbc.exceptions.JDBCDriverException: SAP DBTech JDBC: [257]: sql syntax error: incorrect syntax near ".": line 1 col 306 (at pos 306)
```


col 306 is the following
`"products"."db::Products"."Customers"."ShippingAddress"` 


I tried with 0.13.3 and it's working - the statement that is built is the following:
`CREATE COLUMN TABLE "XSK_SAMPLES_PRODUCTS"."products.db::Products.Customers" ( "Id" NVARCHAR (32) PRIMARY KEY , "Username" NVARCHAR (100) , "FirstName" NVARCHAR (500) , "LastName" NVARCHAR (500) , "Phone" NVARCHAR (200) , "Email" NVARCHAR (300) , "ShippingAddress.Id" NVARCHAR (32) , CONSTRAINT "products.db::Products.Customers.ShippingAddress" FOREIGN KEY ( "ShippingAddress.Id" ) REFERENCES "XSK_SAMPLES_PRODUCTS"."products.db::Products.ShippingAddresses"( "Id" ))`

--- a/modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/processors/table/TableBuilder.java
+++ b/modules/engines/engine-hdb/src/main/java/com/sap/xsk/hdb/ds/processors/table/TableBuilder.java
@@ -160,8 +160,7 @@ public class TableBuilder {
         String foreignKeyName = foreignKey.getName();
         String foreignKeyReferencedTable = foreignKey.getReferencedTable();
         if (caseSensitive) {
-          XSKHDBUtils.escapeArtifactName(foreignKeyName);
-
+          foreignKeyName = XSKHDBUtils.escapeArtifactName(foreignKeyName);
           foreignKeyReferencedTable = XSKHDBUtils.escapeArtifactName(foreignKeyReferencedTable);
         }
         String[] foreignKeyColumns = this.getEscapedColumns(foreignKey.getColumns());

